airbyte-integrations/connectors/source-harvest/unit_tests/integration/test_contacts.py

```diff
@@ -0,0 +1,186 @@
+        )
+
+        source = get_source(config=config)
+        catalog = CatalogBuilder().with_stream(_STREAM_NAME, SyncMode.full_refresh).build()
+        output = read(source, config=config, catalog=catalog, expecting_exception=False)
+
+        assert len(output.records) == 0
+        log_messages = [log.log.message for log in output.logs]
+        assert any("This is most likely due to insufficient permissions" in msg for msg in log_messages)
+
+    @HttpMocker()
+    def test_not_found_error_handling(self, http_mocker: HttpMocker) -> None:
+        """Test that connector ignores 404 errors per manifest config."""
+        config = ConfigBuilder().with_account_id(_ACCOUNT_ID).with_api_token(_API_TOKEN).build()
+
+        http_mocker.get(
+            HarvestRequestBuilder.contacts_endpoint(_ACCOUNT_ID, _API_TOKEN)
+            .with_per_page(50)
+            .with_updated_since("2021-01-01T00:00:00Z")
+            .build(),
+            HttpResponse(body=json.dumps({"error": "not_found"}), status_code=404),
+        )
+
+        source = get_source(config=config)
+        catalog = CatalogBuilder().with_stream(_STREAM_NAME, SyncMode.full_refresh).build()
+        output = read(source, config=config, catalog=catalog, expecting_exception=False)
+
+        assert len(output.records) == 0
+        log_messages = [log.log.message for log in output.logs]
+        assert any("Please ensure that your account ID is properly set" in msg for msg in log_messages)
+
+    @HttpMocker()
+    def test_incremental_sync_with_state(self, http_mocker: HttpMocker) -> None:
+        """Test incremental sync with state."""
+        config = (
+            ConfigBuilder()
+            .with_account_id(_ACCOUNT_ID)
+            .with_api_token(_API_TOKEN)
+            .with_replication_start_date(datetime(2024, 1, 1, tzinfo=timezone.utc))
+            .build()
+        )
+        state = StateBuilder().with_stream_state(_STREAM_NAME, {"updated_at": "2024-01-01T00:00:00Z"}).build()
+
+        http_mocker.get(
+            HarvestRequestBuilder.contacts_endpoint(_ACCOUNT_ID, _API_TOKEN)
+            .with_per_page(50)
+            .with_updated_since("2024-01-01T00:00:00Z")
+            .build(),
+            HttpResponse(
+                body=json.dumps(
+                    {
+                        "contacts": [{"id": 9001, "created_at": "2024-01-02T10:00:00Z", "updated_at": "2024-01-02T10:00:00Z"}],
+                        "per_page": 50,
+                        "total_pages": 1,
+                        "page": 1,
+                        "links": {},
+                    }
+                ),
+                status_code=200,
+            ),
+        )
+
+        source = get_source(config=config, state=state)
+        catalog = CatalogBuilder().with_stream(_STREAM_NAME, SyncMode.incremental).build()
+        output = read(source, config=config, catalog=catalog, state=state)
+
+        assert len(output.records) >= 1
```

Comment:
since we control the inputs/outputs, we should be able to assert on an exact number of records right?